src/mlpack/methods/ann/augmented/tasks/copy_impl.hpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * @file copy_impl.hpp
+ * @author Konstantin Sidorov
+ *
+ * Implementation of CopyTask class
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_AUGMENTED_TASKS_COPY_IMPL_HPP
+#define MLPACK_METHODS_AUGMENTED_TASKS_COPY_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "copy.hpp"
+
+#include <iostream>
+
+using std::cerr;
+
+namespace mlpack {
+namespace ann /* Artificial Neural Network */ {
+namespace augmented /* Augmented neural network */ {
+namespace tasks /* Task utilities for augmented */ {
+
+CopyTask::CopyTask(int maxLength, int nRepeats) :
+    maxLength(maxLength),
+    nRepeats(nRepeats)
+{
+  // Just storing task-specific paramters.
+}
+
+template<typename ModelType>
+double CopyTask::Evaluate(ModelType& model) {
```

Comment:
Maybe it is true, but I am not sure that the score function is the part of the problem. IMHO the task should specify only inputs and their respective outputs. The reason is that there are usually several viable ways to evaluate the distance between predictions and ground truths (e.g., in regression, you can use Euclidean / Manhattan metrics, use L1 / L2 regularizer with different degrees of regularization).

Even here, in the case of `CopyTask`, there are two plausible ways to evaluate the performance: bit precision and sequence precision.